src/client.rs

```diff
@@ -36,15 +39,28 @@ impl Client {
         Ok(Self(tor_client))
     }
 
+    fn check_directory(cache_path: &Path) -> Result<()> {
+        if !cache_path.is_dir() {
+            return Err(Error::CacheCorruption("directory cache does not exist").into());
```

Comment:
the `.into()` is necessary to cast it to anyhow::Error but I'm not sure why that is necessary when it is not in `flatfiledirmgr::check_directory`?